topdown/eval.go

```diff
@@ -1844,33 +1854,52 @@ func (e evalFunc) evalValue(iter unifyIterator, argCount int, findOne bool) erro
 
 	var prev *ast.Term
```

Comment:
❓ Does the `cacheKey` ever matter? IIRC it affects whether we're evaluating the full extent (because the query is `data.foo[_]`) or a sub-thing (query like `a := "bar", data.foo[a]`)

Reply:
The cache matters for EE in that if we get a hit, we won't evaluate that func, so if we include the lookup in EE suppression, we'll always be suppressing any EE for the outer scope, not for the func being called.

The cache key itself isn't important, it's the `err` or `hit` that are.
But maybe you're asking something completely different, an I'm just misinterpreting 😅.